### PR TITLE
prevent tagging validator release without updating version in miner.erl

### PR DESCRIPTION
**Pull request by jeffgrunewald**
as long as we're tagging our releases with `tag-release.sh` and we don't plan to change the erlang versioning scheme or the tagging strategy for validator tags this should prevent us from publishing a validator release without first updating the version function in `src/miner.erl`

### CODE REVIEW

**Review by jadeallenx — APPROVED** [review]

File: tag-release.sh [hunk 1]
```diff
@@ -13,6 +13,30 @@ dep_hash() {
...
+    fi
+}
+
+val_tag() {
+    local VAL_VERSION
+    local MAJ
+    local MIN
+    local PATCH
+
+    VAL_VERSION=$(grep -A 1 "%% MMMmmmPPPP" src/miner.erl | grep -o -E '[0-9]+')
```
Comment (evanmcc) [inline tag-release.sh:32]: can you add a comment in `src/miner.erl` to the effect that we rely on that comment in the tagging script?

**Comment by madninja** [comment]
Hell yes. Take care of Evan’s comment please?

File: tag-release.sh (hunk 1 above)
Comment (jeffgrunewald) [reply tag-release.sh]: i can do this as well, but i tried changing the pattern to match against the structure of the erlang `version/0` function itself, so as long as no one writes any comments that end the line in a period just maintaining a proper erlang function wrapped around that version number should work

**Comment by anthonyra** [comment]
I believe your `strip_zeros` function would return `0010 -> 1` a quick search suggests to use something like `echo $((10#$1))` instead

**Comment by jeffgrunewald** [comment]
> I believe your `strip_zeros` function would return `0010 -> 1` a quick search suggests to use something like `echo $((10#$1))` instead

great catch; didn't test against versions with trailing zeros. thanks @anthonyra !

File: tag-release.sh (hunk 1 above)
Comment (jeffgrunewald) [reply tag-release.sh]: any more thoughts on this vs. updating the comment to add some weight to it @evanmcc ?

**Review by Vagabond — APPROVED** [review]

**Pull request merged**